subsys/bluetooth/host/conn.c

```diff
@@ -37,6 +37,24 @@ NET_BUF_POOL_DEFINE(acl_tx_pool, CONFIG_BT_L2CAP_TX_BUF_COUNT,
 		    BT_L2CAP_BUF_SIZE(CONFIG_BT_L2CAP_TX_MTU),
 		    CONFIG_BT_L2CAP_TX_USER_DATA_SIZE, NULL);
 
+#if BT_L2CAP_TX_FRAG_COUNT > 0
+
+#if defined(BT_CTLR_TX_BUFFER_SIZE)
+#define FRAG_SIZE BT_L2CAP_BUF_SIZE(BT_CTLR_TX_BUFFER_SIZE - 4)
```

Comment:
What's unclear to me here is how you were setting the fragment size before this patch. Were you using Read_Buffer_Size() to decide whether to use `BT_L2CAP_TX_MTU` or  the controller buffer size? If that's the case, why not share `FRAG_SIZE` for that case?

Reply:
I'm not sure I completely understand what you mean by "share FRAG_SIZE for that case"?

Reply:
Ah, thanks for the explanation, I had misunderstood the original sizing approach.